test/Tests/src/Data/Ordering_Spec.enso

```diff
@@ -93,10 +128,106 @@ spec =
+            ["hello", "ahoj", Number.nan, 3].sort . take 3 . should_equal [3, "ahoj", "hello"]
+            ["hello", "ahoj", Number.nan, 3].sort . at 3 . is_nan . should_be_true
+            [100, Date.new 2020, 50].sort . should_equal [50, 100, Date.new 2020]
+            [100, Nothing, Date.new 2020, 50].sort . should_equal [50, 100, Date.new 2020, Nothing]
+            [3, 2, True, False].sort . should_equal [2, 3, False, True]
+            [3, True, 2, False].sort . should_equal [2, 3, False, True]
+            [Nothing, False].sort . should_equal [False, Nothing]
+
+        Test.specify "should be able to sort any single-element vector without any warnings" <|
+            [Nothing].sort . should_equal [Nothing]
+            expect_no_warns [Nothing].sort
+            [[Nothing]].sort . should_equal [[Nothing]]
+            expect_no_warns [[Nothing]].sort
+            [[1]].sort . should_equal [[1]]
+            expect_no_warns [[1]].sort
+
+        Test.specify "should produce warnings when sorting nested vectors" <|
+            [[1], [2]].sort . should_equal [[1], [2]]
+            [[2], [1]].sort . should_equal [[2], [1]]
+
+        Test.specify "should be able to sort primitive values in atoms" pending=topo_sort_pending <|
+            [Ord.Value Nothing, Ord.Value 20, Ord.Value 10].sort . should_equal [Ord.Value 10, Ord.Value 20, Ord.Value Nothing]
+
+        Test.specify "should produce warnings when sorting primitive values in atoms" pending=topo_sort_pending <|
+            expect_incomparable_warn (Ord.Value 1) (Ord.Value Nothing) [Ord.Value 1, Ord.Value Nothing].sort
+
+        Test.specify "should fail to sort custom incomparable values until topological sorting is implemented" <|
+            [(UPair.Value 1 2), (UPair.Value 3 4)].sort . should_fail_with Incomparable_Values
+
+        Test.specify "should attach warning when trying to sort incomparable values" <|
+            expect_incomparable_warn Nothing Number.nan <| [Nothing, Number.nan].sort on_incomparable=Problem_Behavior.Report_Warning
+            expect_incomparable_warn 1 "hello" <| [1, "hello"].sort on_incomparable=Problem_Behavior.Report_Warning
+
+        Test.specify "should respect previous warnings on a vector" <|
+            Problems.expect_warning "my_warn" <| (Warning.attach "my_warn" [3, 2]) . sort
+            Problems.expect_warning "my_warn" <| (Warning.attach "my_warn" [3, Number.nan]) . sort
+            expect_incomparable_warn 3 Number.nan <| (Warning.attach "my_warn" [3, Number.nan]) . sort on_incomparable=Problem_Behavior.Report_Warning
+
+        Test.specify "should respect previous warnings on vectors" pending="https://github.com/enso-org/enso/issues/6070" <|
+            Problems.expect_warning "my_warn" <| [3, Warning.attach "my_warn" 2].sort
+            expect_incomparable_warn 1 Number.nan [1, Warning.attach "my_warn" Number.nan].sort
+            Problems.expect_warning "my_warn" <| [1, Warning.attach "my_warn" Number.nan].sort
```

Comment:
I appreciate.

I wanted to ask if we can have tests with warnings-attached-values, since they've been giving us some pain recently. But I see these are already there. Awesome!